src/main/java/io/cdap/plugin/gcp/common/CmekUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.common;
+
+import com.google.api.pathtemplate.ValidationException;
+import com.google.cloud.kms.v1.CryptoKeyName;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import com.google.common.base.Strings;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+
+import javax.annotation.Nullable;
+
+
+/**
+ *  Cmek Key Utility class to validate cmek in gcp plugins.
+ */
+public class CmekUtils {
+
+  /**
+   * This method validates the cmek formatted string.
+   *
+   * @param key cmek key formatted string
+   * @param collector  failure collector
+   * @return parsed CryptoKeyName object.
+   */
+  public static CryptoKeyName parseCmekKey(String key, FailureCollector collector) {
```

Comment:
please see other comment. Don't think there is a good reason for any caller to get the raw unvalidated string from GCPConfig